lib/ProductOpener/Events.pm

```diff
@@ -0,0 +1,143 @@
+# This file is part of Product Opener.
+#
+# Product Opener
+# Copyright (C) 2011-2022 Association Open Food Facts
+# Contact: contact@openfoodfacts.org
+# Address: 21 rue des Iles, 94100 Saint-Maur des Fossés, France
+#
+# Product Opener is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+=head1 NAME
+
+ProductOpener::Events - Send events to https://events.openfoodfacts.org
+
+=head1 SYNOPSIS
+
+C<ProductOpener::Events> is used to create events to https://events.openfoodfacts.org
+
+    use ProductOpener::Events qw/:all/;
+
+	# TODO
+
+=head1 DESCRIPTION
+
+See https://github.com/openfoodfacts/openfoodfacts-events
+
+=cut
+
+package ProductOpener::Events;
+
+use ProductOpener::PerlStandards;
+use Exporter qw< import >;
+
+BEGIN {
+	use vars qw(@ISA @EXPORT_OK %EXPORT_TAGS);
+	@EXPORT_OK = qw(
+	  &send_event
+	);    # symbols to export on request
+	%EXPORT_TAGS = (all => [@EXPORT_OK]);
+}
+
+use vars @EXPORT_OK;
+
+use Log::Any qw($log);
+
+use Encode;
+use JSON::PP;
+use LWP::UserAgent;
+use HTTP::Request::Common;
+
+use ProductOpener::Config qw/:all/;
+use ProductOpener::Display qw/display_date_iso/;
+
+=head1 FUNCTIONS
+
+=head2 send_event ( $event_ref )
+
+=head3 Arguments
+
+Arguments are passed through a single hash reference with the following keys:
+
+=head4 event_type - required - string
+
+Type of the event (e.g. "product_edited")
+
+=head4 barcode - required - string
+
+Barcode of the product.
+
+=head4 user id - required
+
+=cut
+
+sub send_event ($event_ref) {
+
+	if ((defined $events_url) and ($events_url ne "") > 0) {
```

Comment:
style: why do you have to put > 0 ?

Reply:
ah, my mistake, I replaced a length($events_url) > 0 by $events_url ne "". The length test is supposed to be faster so we have it in a few places in our code, but I find it more difficult to quickly understand what the intent is (checking that the value is not empty).

Reply:
fixed